### PR TITLE
cloudflare-warp: 2022.02.24 -> 2022.7.421

### PR DESCRIPTION
**Pull request by AStevensTaylor**
###### Description of changes

Updates from changelog.gz
```
cloudflare-warp (2022.7.472) unstable; urgency=medium

  * Fixed issue where OS version and device name may not update in ZT Dashboard
  * Prevent users from brute forcing admin override code
  * Prevent disable-on-wifi feature for ZT customers
  * Add support for environment variables in device posture files paths
  * Prevent flushing all nftable rulesets when WARP is disabled
  * Fix incorrect output in "include-only" mode
  * Fix routing issues when running microk8s
  * Fix issue with wildcard expansion on domain fallback and split tunnel rules
  * Fix issue with `warp-diag feedback` failing to upload
  * Prevent hangs and failures on network traffic when in proxy-mode
  * Prevent log files from growing too large
  * Add "Don't Fragment" bit to encapsulated traffic
  * Ensure our DNS servers stay set if another program tries to change them
  * Add ability to upload device state to API for ZT customers

cloudflare-warp (2022.5.346) unstable; urgency=medium

  * Fixed issue where all nftables rulesets were flushed after warp-cli disconnect
  * Fixed false positives when attempting to detect a captive portal
  * Fixed issue where OS version would not be updated in dash after OS update
  * Added support for DNS over TCP
  * Fixed issue where updated posture checks might not take effect until restart
  * Fixed issue with warp-cli where enable/disable with wifi was allowed in Zero Trust mode
  * Fixed issue where too many DNS requests could result in the following error
    appearing in logs: WARN warp::dns: Shedding DNS load
  * Bug fixes and enhancements

 -- Matt Schulte <mschulte@cloudflare.com> Fri, 17 Jun 2022 15:00:00 -0700

cloudflare-warp (2022.4.235) unstable; urgency=medium

  * Removed periodic tunnel checks
  * Changed `warp-cli connect` to mimic 'warp-cli enable-always-on`
  * Changed `warp-cli disconnect` to mimic `warp-cli disable-always-on`
  * Added a simple taskbar icon to show status
  * Added virtual network switching support via warp-cli (`warp-cli
    get-virtual-networks` and `warp-cli set-virtual-network`).
  * Bug fixes and enhancements

 -- Matt Schulte <mschulte@cloudflare.com> Mon, 18 April 2022 12:00:00 -0700

cloudflare-warp (2022.3.253) unstable; urgency=medium

  * `warp-cli delete` run as root will remove the user for an organization
    even if "Allowed To Leave" is disabled in Zero Trust dashboard.
  * Added `warp-cli disable-connectivity-checks` and `warp-cli
    enabled-connectivity-checks` to control connectivity checks
  * Fixed issue with `warp-cli teams-enroll` on platforms where the default
    browser is installed with snap.
  * Added "Device Posture Only" support

 -- Matt Schulte <mschulte@cloudflare.com> Wed, 40 March 2022 15:30:00 -0700

cloudflare-warp (2022.2.288) unstable; urgency=medium

  * Fixed an issue where the organization name became case sensitive and could
    cause a device to lose registration

 -- Matt Schulte <mschulte@cloudflare.com> Thu, 24 Feb 2022 10:10:00 -0800

cloudflare-warp (2022.2.29) unstable; urgency=medium

  * Fixed issue with warp working on distros with v248 of systemd or newer
  * Fixed issue with device posture application checks detecting running
    processes
  * Now that settings exist in the Zero Trust Dashboard the Client UI should
    behave the same regardless of if you manually joined to a Team or if you
    were forced to by local mdm.xml policy
  * Fixed issue where the WARP Client might not get updated settings when it
    initial starts
  * Fixed issue where the Last Seen value was not updated properly in the Zero
    Trust Dashboard while in "doh" mode
  * Fixed issue where device name was not updated in the Zero trust Dashboard if
    the computer name changed after initial registration
  * Added the ability for daemon to parse mdm.xml file on Linux
  * Shows the Zero Trust Terms of Service when user joins an organization

 -- Matt Schulte <mschulte@cloudflare.com> Wed, 9 Feb 2022 14:07:00 -0800
```

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/cloudflare-warp/default.nix [hunk 1]
```diff
@@ -27,7 +28,7 @@ stdenv.mkDerivation rec {
     runHook preInstall
 
     mv usr $out
-    mv lib $out
+    cp -a lib $out && rm -rf lib
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/cloudflare-warp/default.nix:31]:
```suggestion
    cp -a lib $out
    rm -rf lib
```

**Comment by WolfangAukang** [comment]
I have #168092 open which contains some changes to the application. Could you please add `nftables` and apply the wrap to the warp-svc program as shown [here](https://github.com/NixOS/nixpkgs/blob/ba0e1d31f9c28342c0eb9007e5609e56ed76697d/pkgs/tools/networking/cloudflare-warp/default.nix)?

**Comment by SuperSandro2000** [comment]
Also you must follow the commit naming guidelines in the contributing guide. Semantic commits are not allowed.

**Comment by WolfangAukang** [comment]
[New version has been already merged](https://github.com/NixOS/nixpkgs/pull/193060)

**Pull request closed without merging**